src/app/examples/list/load-on-demand/list-load-on-demand-example.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component } from '@angular/core';
+import { of } from 'rxjs';
+
+import { BaseListComponent } from '../base-list.component';
+import { LoadOnDemandEvent } from './../../../../kirby/components/list/list.event';
+
+@Component({
+  selector: 'kirby-list-load-on-demand-example',
+  templateUrl: './list-load-on-demand-example.component.html',
+  styleUrls: ['./list-load-on-demand-example.component.scss'],
+})
+export class ListLoadOnDemandExampleComponent extends BaseListComponent {
+  private itemCount: number = 0;
+
+  constructor() {
+    super();
+    this.items.push(...this.generateItems());
+  }
+
+  onLoadDemand(loadOnDemandEvent: LoadOnDemandEvent): void {
+    // We end the load more event after 20 items, by sending null to the kirby list.
+    if (this.itemCount <= 20) {
+      // lets make a delay to simulate a HTTP call.
+      setTimeout(() => {
+        this.items.push(...this.generateItems());
```

Comment:
It's important that the example shows the correct way to handle the `loadOnDemand` event:
```suggestion
        this.items = [...this.items, ...this.generateItems()];
```

Reply:
I agree that this is closer to the real world scenario, when using a thing like NgRx.

So by doing this you change the reference of the whole list and all the optimizations `ngFor` gives us. This is for web. I don't know how native is.

But if we should do this we should also provide a `trackBy` to the list.